src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -354,10 +356,20 @@ private async Task<ImmutableArray<Diagnostic>> GetAnalyzerDiagnosticsCoreAsync(I
             Compilation compilation,
             CompilationWithAnalyzersOptions analysisOptions,
             AnalysisScope analysisScope,
+            ImmutableArray<DiagnosticSuppressor> suppressors,
             bool categorizeDiagnostics,
             CancellationToken cancellationToken)
         {
-            var driver = compilation.CreateAnalyzerDriver(analysisScope.Analyzers, new AnalyzerManager(analysisScope.Analyzers), severityFilter: SeverityFilter.None);
+            var analyzers = analysisScope.Analyzers;
+            if (!suppressors.IsEmpty)
+            {
+                // Always provide all the diagnostic suppressors to the driver.
+                // We also need to ensure we are not passing any duplicate suppressor instances.
```

Comment:
Are there any particular tests which will fail if we remove the code for eliminating duplicates here?

Reply:
Also, do we generally expect to determine duplicates on the basis of reference equality? Or does it depend on the analyzer subclass's implementation of equality? I expect basically nobody will override Equals in an analyzer class.

Reply:
> Are there any particular tests which will fail if we remove the code for eliminating duplicates here?

Yes, the added test fails when `withSuppressor` test parameter is true, unless we perform this de-duping.

> Also, do we generally expect to determine duplicates on the basis of reference equality

Yes, we don't expect analyzers to override Equals, so reference equality should be fine.